src/app/clinic-schedule-lib/daily-schedule/daily-schedule-appointments.component.ts

```diff
@@ -84,37 +86,42 @@ export class DailyScheduleAppointmentsComponent implements OnInit, OnDestroy {
     const sub = this.clinicDashboardCacheService.getCurrentClinic()
       .subscribe((location) => {
         this.selectedClinic = location;
-
+        if (this.clinicDashboardCacheService.didLocationChange(location)) {
+          this.loadData();
+        }
       });
 
     this.subs.push(sub);
 
-    // get the current page url and params
+    this.loadData();
```

Comment:
Both calls to loadData() are invoked onInit, something to be worried about?

Reply:
The second call is invoked conditionally, only when the location change

Reply:
Ok